src/all_equal_value_err.rs

```diff
@@ -0,0 +1,29 @@
+#[cfg(feature = "use_std")]
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
+
+/// Value returned for the error case of `Itertools::all_equal_value()`.
```

Comment:
Can you make this an intradoc link:
```suggestion
/// Value returned for the error case of [`Itertools::all_equal_value]`.
```
you might need to add a `#[cfg(doc)] use crate::Itertools;` to the top of the file.

Reply:
Done.